### PR TITLE
Ability do deploy Istio

### PR DESCRIPTION
**Pull request by rpudlowski93**
- Ability to deploy Istio ( control plane + ingress gateway ) using Istio operator
- Tested using example application from official istio documentation
- Possible to deploy istio in offline mode
- Istio implemented in application role
- Dockerfile updated of istioctl binaries (istio version based on istioctl version) 
- Documentations updated (components list, how to, changelog)

### CODE REVIEW

File: core/src/epicli/data/common/defaults/configuration/applications.yml [hunk 1]
```diff
@@ -169,3 +169,18 @@ specification:
         DEFAULT_POOL_SIZE: 25
         RESERVE_POOL_SIZE: 25
         POOL_MODE: transaction
+
+## --- istio ---
+
+  - name: istio
+    enabled: yes
```
Comment (seriva) [inline core/src/epicli/data/common/defaults/configuration/applications.yml:176]: Dont enable by default.

File: core/src/epicli/.devcontainer/Dockerfile [hunk 2]
```diff
@@ -35,6 +36,11 @@ RUN chmod +x /config-pre.sh \
     && chmod +x ./kubectl \
     && mv ./kubectl /usr/local/bin/kubectl \
     && kubectl version --client \
+    && echo "Installing istioctl binary ..." \
```
Comment (seriva) [inline core/src/epicli/.devcontainer/Dockerfile:39]: Also add this to the main [deployment container](https://github.com/epiphany-platform/epiphany/blob/develop/Dockerfile) in the root of the repo

File: docs/home/howto/ISTIO.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+Using this configuration file, controller will detect Istio Operator resource in first of watched namespaces and will install Istio components corresponding to the specified profile (default). Using the default profile, istio control plane and istio ingress gateway will be deployed in istio-system namespace.
+
+### How to set up service mesh for an application
+
+The default istio installation use automcatic sidecar injection. You need to label the namespace where application will be hosted: `kubectl label namespace default istio-injection=enabled`
+
+If the proper namespaces is already labeled and istio is deployed, You can deploy Your applicayion or restart deployment if exists.
+
+You may need to make the application accessible from outside of your Kubernetes cluster. An Istio Gateway which was deployed using default profile is used for this purpose. Define the the ingress gateway deploying gateway and virtual service specification. The gateway specification describes the L4-L6 properties of a load balancer and the virtual service specification describe the L7 properties of a load balancer.
```
Comment (to-bar) [inline docs/home/howto/ISTIO.md:46]:
```suggestion
You may need to make an application accessible from outside of your Kubernetes cluster. An Istio Gateway which was deployed using default profile is used for this purpose. Define the ingress gateway deploying gateway and virtual service specification. The gateway specification describes the L4-L6 properties of a load balancer and the virtual service specification describes the L7 properties of a load balancer.
```

**Review by to-bar — CHANGES_REQUESTED** [review]

**Review by seriva — APPROVED** [review]

**Review by rafzei — DISMISSED** [review]
:+1:

File: docs/home/howto/ISTIO.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      profile: default # Check all possibilites https://istio.io/latest/docs/setup/additional-setup/config-profiles/
+      name: istiocontrolplane
+
+```
+
+Using this configuration file, controller will detect Istio Operator resource in first of watched namespaces and will install Istio components corresponding to the specified profile (default). Using the default profile, Istio control plane and Istio ingress gateway will be deployed in istio-system namespace.
+
+### How to set up service mesh for an application
+
+The default Istio installation use automcatic sidecar injection. You need to label the namespace where application will be hosted:
```
Comment (to-bar) [inline docs/home/howto/ISTIO.md:42]:
```suggestion
The default Istio installation uses automcatic sidecar injection. You need to label the namespace where application will be hosted:
```

**Review by to-bar — CHANGES_REQUESTED** [review]

**Review by to-bar — APPROVED** [review]

**Review by ar3ndt — COMMENTED** [review]
I have met issue during installation @rhel. Seems like package is unavailable: istio/proxyv2:1.8.1

**Comment by przemyslavic** [comment]
@ar3ndt when exactly was that? On Wednesday I was able to install Istio on all environments (`AWS` / `Azure` x `Ubuntu` / `RHEL`).

File: docs/home/howto/ISTIO.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
+## Istio
+
+Open source platform which allows you to run service mesh for distributed microservice architecture. It allows to connect, manage and run secure connections between microservices and brings a lot of features such as load balancing, monitoring and service-to-service authentication without any changes in service code. Read more about Istio [here](https://istio.io/latest/docs/concepts/what-is-istio/).
```
Comment (sk4zuzu) [inline docs/home/howto/ISTIO.md:3]: Isn't this better? :thinking: 
```suggestion
Open source platform which allows you to run service mesh for distributed microservice architecture. It allows to connect, manage and run secure connections between microservices and brings lots of features such as load balancing, monitoring and service-to-service authentication without any changes in service code. Read more about Istio [here](https://istio.io/latest/docs/concepts/what-is-istio/).
```

**Review by sk4zuzu — CHANGES_REQUESTED** [review]

**Comment by ar3ndt** [comment]
> @ar3ndt when exactly was that? On Wednesday I was able to install Istio on all environments (`AWS` / `Azure` x `Ubuntu` / `RHEL`).

Im doing it again - hopefully it was temporary docker hub issue.

**Comment by ar3ndt** [comment]
> > @ar3ndt when exactly was that? On Wednesday I was able to install Istio on all environments (`AWS` / `Azure` x `Ubuntu` / `RHEL`).
> 
> Im doing it again - hopefully it was temporary docker hub issue.

it is fine now.

**Review by seriva — APPROVED** [review]

**Review by sk4zuzu — APPROVED** [review]
:heart_eyes:

**Pull request merged**